provider/scripts/spark/offline_store_spark_runner.py

```diff
@@ -65,14 +66,15 @@ def execute_sql_query(job_type, output_uri, sql_query, spark_configs, source_lis
 
         if job_type == "Transformation" or job_type == "Materialization" or job_type == "Training Set":
             for i, source in enumerate(source_list):
-                if source.endswith(".csv"):
+                file_extension = Path(source).suffix
+                if file_extension == ".csv":
                     source_df = spark.read.option("header","true").option("recursiveFileLookup", "true").csv(source) 
                     source_df.createOrReplaceTempView(f'source_{i}')
-                elif source.endswith(".parquet"):
+                elif file_extension == ".parquet" or file_extension == "":
```

Comment:
Just a minor suggestion from someone (i.e. me) with almost no hands-on experience with Spark: could we store `file_extension == ""` in a variable (e.g. `is_directory`) so that it's clear that a missing file extension denotes a directory.

If you really want to help someone like me out, you might also add a little comment about why a directory works with the same parquet API, but that's up to you to decide if that's far too obvious for the average Spark user.

Reply:
i wholeheartedly agree with this. Remembering that 99% of the time code is being read means that making things as easy to understand is. file_extension == "" makes me think:

Hmm what does it mean that file extension is empty?
Does it mean that's a directory?

When all of that could be short circuited with either a comment or just another function.